test/mpi/dtpools/src/dtpools_custom.c

```diff
@@ -0,0 +1,694 @@
+#include <string.h>
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <ctype.h>
+#include "dtpools_internal.h"
+
+#define SKIP_SPACE while (isspace(*s)) s++
+#define EXPECT_DIGIT \
+    do { \
+        while (isspace(*s)) s++; \
+        if (!isdigit(*s)) { \
+            DTPI_ERR_ASSERT(0, rc); \
+        } \
+    } while (0)
+#define EXPECT_ALPHA \
+    do { \
+        while (isspace(*s)) s++; \
+        if (!isalpha(*s)) { \
+            DTPI_ERR_ASSERT(0, rc); \
+        } \
+    } while (0)
+#define EXPECT_CHAR(c) \
+    do { \
+        while (isspace(*s)) s++; \
+        if (*s != (c)) { \
+            DTPI_ERR_ASSERT(0, rc); \
+        } \
+    } while (0)
+#define SKIP_DIGIT \
+    do { \
+        while (*s && isdigit(*s)) s++; \
+        DTPI_ERR_ASSERT(*s, rc); \
+    } while (0)
+#define SKIP_ALPHA \
+    do { \
+        while (*s && isalpha(*s)) s++; \
+        DTPI_ERR_ASSERT(*s, rc); \
+    } while (0)
+#define SKIP_UNTIL(c) \
+    do { \
+        while (*s && *s != (c)) s++; \
+        DTPI_ERR_ASSERT(*s, rc); \
+    } while (0)
+
+int DTPI_parse_desc(const char *s, const char **parts, int *num_parts, int max_parts)
```

Comment:
the custom description is the same as the one printed for debug purposes? or is the format different?

Reply:
I tried to make it the same so one could simply copy-paste the debug output into a custom test. But the custom function will accept more flexible inputs. For example, it doesn't care about line breaks and the base type.

Reply:
OK, I see your comment in the PR description now. Looks good.